src/go/rdctl/pkg/autostart/autostart.go

```diff
@@ -0,0 +1,98 @@
+package autostart
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+const regPath = `C:\Windows\system32\reg.exe`
+
+// Ensures that Rancher Desktop is configured to start at login (or not).
+func EnsureAutostart(autostartDesired bool) error {
+	switch runtime.GOOS {
+	case "windows":
+		return ensureAutostartWindows(autostartDesired)
+	case "darwin":
+		return nil
+	case "linux":
+		return nil
+	default:
+		return fmt.Errorf(`not supported for platform "%s"`, runtime.GOOS)
+	}
```

Comment:
Instead of this switch statement you should create separate `autostart_windows.go`, `autostart_linux.go`, and `autostart_darwin.go` files that each have a platform-specific `EnsureAutostart` implementation.

Go will only compile `*-$ARCH.go` files when targing that arch, so only one of the 3 files will be compiled and linked.

Reply:
:+1: making this change.